CHERRY-PICK: fix: mitigate race conditions on community

osmaczko opened the pull request:
CHERRY-PICK OF: https://github.com/status-im/status-go/pull/5060

osmaczko committed:
fix_: mitigate race conditions on community

mitigates: status-im/status-desktop#14432

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (4)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | 7b5879059d3e54213778865c6329ef9254be271d | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5063/1/) | 2024-04-17 07:38:20 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5063/1//consoleText) |
| :heavy_check_mark: | 7b5879059d3e54213778865c6329ef9254be271d | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5063/1/) | 2024-04-17 07:41:20 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240417-073653-7b5879059d3e54213778865c6329ef9254be271d-pr5063.zip) |
| :heavy_check_mark: | 7b5879059d3e54213778865c6329ef9254be271d | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5063/1/) | 2024-04-17 07:42:10 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240417-073652-7b5879059d3e54213778865c6329ef9254be271d-pr5063.zip) |
| :heavy_check_mark: | 7b5879059d3e54213778865c6329ef9254be271d | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5063/1/) | 2024-04-17 07:42:30 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240417-073651-7b5879059d3e54213778865c6329ef9254be271d-pr5063.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | b004d888633e44781399da9bdf21b56a06a06b7a | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5063/2/) | 2024-04-17 12:10:05 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240417-120212-b004d888633e44781399da9bdf21b56a06a06b7a-pr5063.zip) |
| :heavy_check_mark: | b004d888633e44781399da9bdf21b56a06a06b7a | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5063/2/) | 2024-04-17 12:11:01 | ~8 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240417-120208-b004d888633e44781399da9bdf21b56a06a06b7a-pr5063.aar) |
| :heavy_check_mark: | b004d888633e44781399da9bdf21b56a06a06b7a | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5063/2/) | 2024-04-17 12:34:05 | ~31 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240417-120208-b004d888633e44781399da9bdf21b56a06a06b7a-pr5063.zip) |
| :heavy_multiplication_x: | b004d888633e44781399da9bdf21b56a06a06b7a | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5063/2/) | 2024-04-17 12:36:06 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5063/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 728293103f0374752b841c77eb202f2967f709a0 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5063/3/) | 2024-04-17 14:27:29 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240417-142141-728293103f0374752b841c77eb202f2967f709a0-pr5063.zip) |
| :heavy_check_mark: | 728293103f0374752b841c77eb202f2967f709a0 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5063/3/) | 2024-04-17 14:28:44 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240417-142141-728293103f0374752b841c77eb202f2967f709a0-pr5063.aar) |
| :heavy_check_mark: | 728293103f0374752b841c77eb202f2967f709a0 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5063/3/) | 2024-04-17 14:29:34 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240417-142141-728293103f0374752b841c77eb202f2967f709a0-pr5063.zip) |
| :heavy_check_mark: | 728293103f0374752b841c77eb202f2967f709a0 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5063/3/) | 2024-04-17 15:14:04 | ~52 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5063/3//consoleText) |

osmaczko committed:
fix_: keep community locks map unreleased when Manager stops

`ReevaluateMembers` is run as a separate goroutine and sometimes it is
executed after `Manager` has been stopped. It tries to use the lock and
in consequence, it panics. Ensuring the map is still there prevents that.

^Happened in test: `TestCreateTokenPermission`.
